meta/sai_meta.cpp

```diff
@@ -0,0 +1,4884 @@
+}
+
+void set_object(
+        _In_ const sai_object_meta_key_t meta_key,
+        _In_ const sai_attr_metadata_t& md,
+        _In_ const sai_attribute_t *attr)
+{
+    SWSS_LOG_ENTER();
+
+    std::string key = get_object_meta_key_string(meta_key);
+
+    if (!object_exists(key))
+    {
+        SWSS_LOG_ERROR("FATAL: object %s don't exists", key.c_str());
+        throw;
+    }
+
+    META_LOG_DEBUG(md, "set attribute %d on %s", attr->id, key.c_str());
+
+    auto p = new SaiAttrWrapper(meta_key.object_type, md.serializationtype, *attr);
+
+    ObjectAttrHash[key][attr->id] = std::shared_ptr<SaiAttrWrapper>(p);
+}
+
+const std::vector<std::shared_ptr<SaiAttrWrapper>> get_object(
+        _In_ const sai_object_meta_key_t meta_key)
+{
+    std::string key = get_object_meta_key_string(meta_key);
+
+    if (!object_exists(key))
+    {
+        SWSS_LOG_ERROR("FATAL: object %s don't exists", key.c_str());
+        throw;
+    }
+
+    std::vector<std::shared_ptr<SaiAttrWrapper>> attrs;
+
+    const auto& hash = ObjectAttrHash[key];
+
+    for (auto it = hash.begin(); it != hash.end(); ++it)
+    {
+        attrs.push_back(it->second);
+    }
+
+    return attrs;
+}
+
+void remove_object(
+        _In_ const sai_object_meta_key_t meta_key)
+{
+    SWSS_LOG_ENTER();
+
+    std::string key = get_object_meta_key_string(meta_key);
+
+    if (!object_exists(key))
+    {
+        SWSS_LOG_ERROR("FATAL: object %s don't exists", key.c_str());
+        throw;
+    }
+
+    SWSS_LOG_DEBUG("removing object %s", key.c_str());
+
+    ObjectAttrHash.erase(key);
+}
+
+void create_object(
+        _In_ const sai_object_meta_key_t& meta_key)
+{
+    SWSS_LOG_ENTER();
+
+    std::string key = get_object_meta_key_string(meta_key);
+
+    if (object_exists(key))
+    {
+        SWSS_LOG_ERROR("FATAL: object %s already exists", key.c_str());
+        throw;
+    }
+
+    SWSS_LOG_DEBUG("creating object %s", key.c_str());
+
+    ObjectAttrHash[key] = {};
+}
+
+sai_status_t meta_generic_validation_objlist(
+        _In_ const sai_attr_metadata_t& md,
+        _In_ uint32_t count,
+        _In_ const sai_object_id_t* list)
+{
+    SWSS_LOG_ENTER();
+
+    if (count > MAX_LIST_COUNT)
+    {
+        META_LOG_ERROR(md, "object list count %u is > then max list count %u", count, MAX_LIST_COUNT);
+
+        return SAI_STATUS_INVALID_PARAMETER;
+    }
+
+    if (list == NULL)
+    {
+        if (count == 0)
+        {
+            return SAI_STATUS_SUCCESS;
+        }
+
+        META_LOG_ERROR(md, "object list is null, but count is %u", count);
+
+        return SAI_STATUS_INVALID_PARAMETER;
+    }
+
+    std::set<sai_object_id_t> oids;
+
+    sai_object_type_t object_type = SAI_OBJECT_TYPE_NULL;
+
+    for (uint32_t i = 0; i < count; ++i)
+    {
+        sai_object_id_t oid = list[i];
+
+        if (oids.find(oid) != oids.end())
+        {
+            META_LOG_ERROR(md, "object on list [%u] oid 0x%llx is duplicated, but not allowed", i, oid);
```

Comment:
@kcudnik Can you please provide some context on this check?

Reply:
this checks whether duplicated oids are on the list of the parameters, this is fdom 2016 code, why intrest on it now?

Reply:
I was trying to find some context on this check
https://github.com/sonic-net/sonic-sairedis/blob/master/meta/Meta.cpp#L6188
and followed the git history to get the original PR.
I was trying to get some reasoning behind this check. Is it invalid for an OID to be repeated in an OID list?

Reply:
it make no sense to repeat oid on the list